src/app/o/[orgId]/events/[eventId]/layout.tsx

```diff
@@ -0,0 +1,57 @@
+import { FC, PropsWithChildren } from 'react';
+import { Metadata } from 'next';
+import { headers } from 'next/headers';
+import { notFound } from 'next/navigation';
+
+import HomeThemeProvider from 'features/home/components/HomeThemeProvider';
+import PublicEventLayout from 'features/organizations/layouts/PublicEventLayout';
+import BackendApiClient from 'core/api/client/BackendApiClient';
+import { ZetkinEvent } from 'utils/types/zetkin';
+
+type Props = PropsWithChildren<{
+  params: {
+    eventId: number;
+    orgId: number;
+  };
+}>;
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const headersList = headers();
+  const headersEntries = headersList.entries();
+  const headersObject = Object.fromEntries(headersEntries);
+  const apiClient = new BackendApiClient(headersObject);
+
+  const event = await apiClient.get<ZetkinEvent>(
+    `/api/orgs/${params.orgId}/actions/${params.eventId}`
+  );
+
+  return {
+    icons: [{ url: '/logo-zetkin.png' }],
+    //TODO: Fallback on something for untitled events
+    title: event.title || event.activity?.title,
```

Comment:
It should be able to use the backend `getMessages()` function to get the correct messages for the current locale.